libs/template/renderer.go

```diff
@@ -0,0 +1,260 @@
+package template
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"path"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/databricks/cli/libs/filer"
+	"github.com/databricks/cli/libs/log"
+	"github.com/databricks/databricks-sdk-go/logger"
+	"golang.org/x/exp/slices"
+)
+
+type inMemoryFile struct {
+	// Unix like path for the file (using '/' as the separator)
+	path    string
+	content []byte
+}
+
+// Renders a databricks template as a project
+type renderer struct {
+	ctx context.Context
+
+	// A config that is the "dot" value available to any template being rendered.
+	// Refer to https://pkg.go.dev/text/template for how templates can use
+	// this "dot" value
+	config map[string]any
+
+	// A base template with helper functions and user defined template in the
+	// library directory loaded. This is used as the base to compute any project
+	// templates during file tree walk
```

Comment:
"... This is cloned for each...". I was confused reading base because it implies extending.